### DISCUSSION TITLE
Disable auto-staging?

### DISCUSSION THREAD

**Discussion by lonix1** [jesseduffield/lazygit#4102] [Q&A]
I often stage changes in my IDE, then go to Lazygit to perform an amend. But sometimes Lazygit is not fast enough in detecting the changes, and shows:
> You have not staged any files. Commit all files?

It's too easy to make a mistake (or not notice) and press enter; to untangle that mess is often difficult.

I don't want auto-staging behaviour. Does such a setting exist?

**Comment by stefanhaller** [comment]
No, there is no such setting, but it would be easy to add.

I'm curious though, are you on Windows? As far as I know, that's the only platform that doesn't support the focus events that lazygit uses to refresh the files panel when it becomes active. When focus events work, refreshing should be fast enough that it shouldn't be an issue.

**Reply by lonix1** [reply]
No I'm using Linux.

In my experience it detects the changes only sometimes. I often have to press <kbd>r</kbd> to refresh the panel, or wait a while (in which case I often mash the up/down keys to force a state change).

**Reply by stefanhaller** [reply]
Interesting. What terminal emulator are you using? Maybe it doesn't support focus events. Without focus events, the files panel refreshes every 10 seconds by default, which would explain what you are seeing.

**Reply by lonix1** [reply]
I'm using Ubuntu, so I guess it's `gnome-terminal`.

Edit: I use Lazygit inside tmux; unsure whether that matters?

**Comment by migig** [comment]
I hoping for this setting is to be implemented. Also I sometime make this mistake - I press "Enter" but did not read the message.
